docs/source/guide/artifacts/index.rst

```diff
@@ -0,0 +1,16 @@
+Artifacts
+=========
+
+In LineaPy, an artifact refers to any intermediate result from the development process. Most often, an artifact
```

Comment:
I'd get @dorx to read this too. This is an important enough concept to Linea's core product to get her eyes on this file.

Reply:
This is actually text from our existing [Concepts](https://docs.lineapy.org/en/latest/fundamentals/concepts.html) page. I will ask Doris to check it to see if it is outdated.

Reply:
sgtm

Reply:
Confirmed with @dorx that text in `Concepts` page needs revamp eventually but this does not block merging of this PR.